lightning/src/onion_message/messenger.rs

```diff
@@ -977,19 +1007,30 @@ where
 		)
 	}
 
-	fn handle_onion_message_response<T: OnionMessageContents>(
+	/// Handles the response to an [`OnionMessage`] based on its [`ResponseInstruction`],
+	/// enqueueing any response for sending.
+	pub fn handle_onion_message_response<T: OnionMessageContents> (
 		&self, response: ResponseInstruction<T>
-	) {
-		if let ResponseInstruction::WithoutReplyPath(response) = response {
-			let message_type = response.message.msg_type();
-			let _ = self.find_path_and_enqueue_onion_message(
-				response.message, Destination::BlindedPath(response.reply_path), None,
-				format_args!(
-					"when responding with {} to an onion message with path_id {:02x?}",
-					message_type,
-					response.path_id
-				)
-			);
+	) -> Result<Option<SendSuccess>, SendError> {
+		let (response, reply_path) = match response {
+			ResponseInstruction::WithReplyPath(response) => (response, self.create_blinded_path().ok()),
+			ResponseInstruction::WithoutReplyPath(response) => (response, None),
+			ResponseInstruction::NoResponse => return Ok(None),
+		};
+
+		let message_type = response.message.msg_type();
+		let res = self.find_path_and_enqueue_onion_message(
+			response.message, Destination::BlindedPath(response.reply_path), reply_path,
+			format_args!(
+				"when responding to {} onion message with path_id {:02x?}",
+				message_type,
+				response.path_id
+			)
+		);
+
+		match res {
+			Ok(result) => Ok(Some(result)),
+			Err(result) => Err(result)
 		}
```

Comment:
Use `Option::map` for brevity:

```rust
self.find_path_and_enqueue_onion_message(
    // ...
).map(|result| Some(result))
```

Reply:
Done! [2d769d6](https://github.com/lightningdevkit/rust-lightning/commit/2d769d683b2ae9de8d04c23c6bf24684b902eac4)
Thanks for the pointer!